core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -88,11 +88,22 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
 
   def map[D](f: B => D)(implicit F: Functor[F]): EitherT[F, A, D] = bimap(identity, f)
 
+  def mapF[N[_], D](fe: F[Either[A, B]] => N[Either[A, D]]): EitherT[N, A, D] = EitherT(fe(value))
+
+  def recoverF[E](pf: PartialFunction[E, Either[A, B]])(implicit ae: ApplicativeError[F, E]): EitherT[F, A, B] =
```

Comment:
So as @djspiewak pointed out in #1643, there are two ways to provide a `MonadError` instance for a `EitherT`, the existing one ignores the error in `F` and recovers from the left side of the Either. We can have a new `MonadError` instance if the`F` already has `MonadError` instance and delegate to that, which is basically what you are doing with `recoverF`. These two instances might be able to coexist since the Error type is different, which should be true as long as the E in `F: MonadError` and the `EitherT`'s left type are different, right?